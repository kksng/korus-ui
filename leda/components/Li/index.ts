import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';

export interface LiProps extends React.HTMLAttributes<HTMLLIElement> {
  ref?: React.Ref<LiRefCurrent>,
  shouldRender?: boolean,
  [x: string]: unknown,
}

export interface LiRefCurrent {
  wrapper: HTMLLIElement | null,
}

export const Li = htmlTagFactory('Li') as React.FC<LiProps>;
