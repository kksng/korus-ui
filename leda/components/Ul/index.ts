import * as React from 'react';
import { htmlTagFactory } from '../../src/HTMLTagsFactory';

export interface UlProps extends React.HTMLAttributes<HTMLUListElement> {
  ref?: React.Ref<UlRefCurrent>,
  shouldRender?: boolean,
  [x: string]: unknown,
}

export interface UlRefCurrent {
  wrapper: HTMLUListElement | null,
}

export const Ul = htmlTagFactory('Ul') as React.FC<UlProps>;
