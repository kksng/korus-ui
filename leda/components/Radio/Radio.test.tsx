// @ts-nocheck
import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { RadioGroup, RadioButton } from './index';
import { Ul } from '../Ul';
import { Li } from '../Li';

describe('Radio SNAPSHOTS', () => {
  it('should render', () => {
    const radio = (
      <RadioGroup value="radio-1" onChange={jest.fn()} name="radio-group">
        <RadioButton value="radio-1" id="radio-1">1</RadioButton>
        <RadioButton value="radio-2" id="radio-2">2</RadioButton>
        <RadioButton value="radio-3" id="radio-3">3</RadioButton>
      </RadioGroup>
    );
    const wrapper = mount(radio);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render controllable mode', () => {
    const radio = (
      <RadioGroup value="radio-1" onChange={jest.fn()} name="radio-group">
        <RadioButton value="radio-1" id="radio-1">1</RadioButton>
        <RadioButton value="radio-2" id="radio-2">2</RadioButton>
        <RadioButton value="radio-3" id="radio-3">3</RadioButton>
      </RadioGroup>
    );
    const wrapper = mount(radio);

    expect(wrapper.find('input').first().props().checked).toBeTruthy();

    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.setProps({ value: 'radio-3' });

    expect(wrapper.find('input').first().props().checked).toBeFalsy();

    expect(wrapper.find('input').last().props().checked).toBeTruthy();

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should default to value of the first radio', () => {
    const radio = (
      <RadioGroup name="radio-group">
        <RadioButton value="radio-1" id="radio-1">1</RadioButton>
        <RadioButton value="radio-2" id="radio-2">2</RadioButton>
        <RadioButton value="radio-3" id="radio-3">3</RadioButton>
      </RadioGroup>
    );
    const wrapper = mount(radio);

    expect(wrapper.find('input').first().props().checked).toBeTruthy();
  });

  describe('different component states', () => {
    it('should render checked', () => {
      const radio = (
        <RadioGroup value="radio-1" onChange={jest.fn()} name="radio-group">
          <RadioButton onChange={jest.fn()} id="radio-1" value="radio-1">1</RadioButton>
        </RadioGroup>
      );
      const wrapper = mount(radio);

      expect(wrapper.find('input').first().props().checked).toBeTruthy();

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render disabled', () => {
      const radio = (
        <RadioGroup value="radio-1" onChange={jest.fn()} name="radio-group">
          <RadioButton value="radio-1" id="radio-1" isDisabled>1</RadioButton>
          <RadioButton value="radio-2" id="radio-2">2</RadioButton>
          <RadioButton value="radio-3" id="radio-3">3</RadioButton>
        </RadioGroup>
      );
      const wrapper = mount(radio);

      expect(wrapper.find('input').first().props().disabled).toBeTruthy();

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});

describe('Radio HANDLERS', () => {
  it('should trigger onChange', () => {
    const onChange = jest.fn();
    const radio = (
      <RadioGroup value="radio-1" onChange={onChange} name="radio-group">
        <RadioButton value="radio-1" id="radio-1">1</RadioButton>
        <RadioButton value="radio-2" id="radio-2">2</RadioButton>
        <RadioButton value="radio-3" id="radio-3">3</RadioButton>
      </RadioGroup>
    );
    const wrapper = mount(radio);

    wrapper.find('input').last().props().onChange({ component: { value: 'radio-3' } });

    expect(onChange).toHaveBeenCalled();
  });

  it('should have correct event format', () => {
    const onChange = jest.fn();
    const radio = (
      <RadioGroup value="radio-1" onChange={onChange} name="radio-group">
        <RadioButton value="radio-1" id="radio-1">1</RadioButton>
        <RadioButton value="radio-2" id="radio-2">2</RadioButton>
        <RadioButton value="radio-3" id="radio-3">3</RadioButton>
      </RadioGroup>
    );
    const wrapper = mount(radio);

    wrapper.find('input').last().getDOMNode().addEventListener('change', (ev) => wrapper.props().onChange(ev));

    wrapper.find('input').last().props().onChange({ value: 'radio-3', name: 'radio-group' });

    expect(onChange).toHaveBeenCalled();

    const [[event]] = onChange.mock.calls;

    expect(event.component.value).toEqual('radio-3');

    expect(event.component.name).toEqual('radio-group');
  });
});

describe('Radio ATTRIBUTES', () => {
  it('should be wrapped with given wrapper', () => {
    const radio = (
      <RadioGroup wrapperRender={({ elementProps }) => <Ul {...elementProps} />} value="radio-1" onChange={jest.fn()} name="radio-group">
        <RadioButton wrapperRender={() => <Li />} value="radio-1" id="radio-1">1</RadioButton>
        <RadioButton wrapperRender={() => <Li />} value="radio-2" id="radio-2">2</RadioButton>
        <RadioButton wrapperRender={() => <Li />} value="radio-3" id="radio-3">3</RadioButton>
      </RadioGroup>
    );
    const wrapper = mount(radio);

    expect(wrapper.find('ul')).toHaveLength(1);

    expect(wrapper.find('li')).toHaveLength(3);
  });

  it('should give label input id', () => {
    const radio = (
      <RadioGroup wrapperRender={({ elementProps }) => <Ul {...elementProps} />} value="radio-1" onChange={jest.fn()} name="radio-group">
        <RadioButton wrapperRender={({ elementProps }) => <Li {...elementProps} />} id="radio-1" value="radio-1">1</RadioButton>
        <RadioButton wrapperRender={({ elementProps }) => <Li {...elementProps} />} id="radio-2" value="radio-2">2</RadioButton>
        <RadioButton wrapperRender={({ elementProps }) => <Li {...elementProps} />} id="radio-3" value="radio-3">3</RadioButton>
      </RadioGroup>
    );
    const wrapper = mount(radio);

    expect(wrapper.find('label').first().props().htmlFor).toEqual('radio-1');

    expect(wrapper.find('label').last().props().htmlFor).toEqual('radio-3');
  });

  it('should accept className', () => {
    const radio = (
      <RadioGroup _box wrapperRender={({ elementProps }) => <Ul {...elementProps} />} value="radio-1" onChange={jest.fn()} name="radio-group">
        <RadioButton _active wrapperRender={({ elementProps }) => <Li {...elementProps} />} value="radio-1" id="radio-1">1</RadioButton>
        <RadioButton wrapperRender={({ elementProps }) => <Li {...elementProps} />} value="radio-2" id="radio-2">2</RadioButton>
        <RadioButton className="custom" wrapperRender={({ elementProps }) => <Li {...elementProps} />} value="radio-3" id="radio-3">3</RadioButton>
      </RadioGroup>
    );
    const wrapper = mount(radio);

    expect(wrapper.find('Ul').hasClass('box')).toBeTruthy();

    expect(wrapper.find('label').first().hasClass('active')).toBeTruthy();

    expect(wrapper.find('label').last().hasClass('custom')).toBeTruthy();
  });
});
