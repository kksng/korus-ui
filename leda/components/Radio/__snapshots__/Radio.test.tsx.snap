// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio SNAPSHOTS different component states should render checked 1`] = `
<RadioGroup
  name="radio-group"
  onChange={[MockFunction]}
  value="radio-1"
>
  <Wrapper
    className="radio-wrapper"
  >
    <Div
      className="radio-wrapper"
    >
      <div
        className="radio-wrapper"
      >
        <RadioButton
          id="radio-1"
          isChecked={true}
          key=".0"
          name="radio-group"
          onChange={[Function]}
          theme={
            Object {
              "input": "radio-button-input",
              "item": "radio-button",
              "label": "radio-button-label",
              "wrapper": "radio-wrapper",
            }
          }
          value="radio-1"
        >
          <Wrapper
            className="radio-button"
          >
            <Div
              className="radio-button"
            >
              <div
                className="radio-button"
              >
                <Input
                  checked={true}
                  className="radio-button-input"
                  id="radio-1"
                  onChange={[Function]}
                  type="radio"
                  value="radio-1"
                >
                  <input
                    checked={true}
                    className="radio-button-input"
                    id="radio-1"
                    onChange={[Function]}
                    type="radio"
                    value="radio-1"
                  />
                </Input>
                <label
                  className="radio-button-label"
                  htmlFor="radio-1"
                >
                  1
                </label>
              </div>
            </Div>
          </Wrapper>
        </RadioButton>
      </div>
    </Div>
  </Wrapper>
</RadioGroup>
`;

exports[`Radio SNAPSHOTS different component states should render disabled 1`] = `
<RadioGroup
  name="radio-group"
  onChange={[MockFunction]}
  value="radio-1"
>
  <Wrapper
    className="radio-wrapper"
  >
    <Div
      className="radio-wrapper"
    >
      <div
        className="radio-wrapper"
      >
        <RadioButton
          id="radio-1"
          isChecked={true}
          isDisabled={true}
          key=".0"
          name="radio-group"
          onChange={[Function]}
          theme={
            Object {
              "input": "radio-button-input",
              "item": "radio-button",
              "label": "radio-button-label",
              "wrapper": "radio-wrapper",
            }
          }
          value="radio-1"
        >
          <Wrapper
            className="radio-button"
          >
            <Div
              className="radio-button"
            >
              <div
                className="radio-button"
              >
                <Input
                  checked={true}
                  className="radio-button-input"
                  disabled={true}
                  id="radio-1"
                  onChange={[Function]}
                  type="radio"
                  value="radio-1"
                >
                  <input
                    checked={true}
                    className="radio-button-input"
                    disabled={true}
                    id="radio-1"
                    onChange={[Function]}
                    type="radio"
                    value="radio-1"
                  />
                </Input>
                <label
                  className="radio-button-label"
                  htmlFor="radio-1"
                >
                  1
                </label>
              </div>
            </Div>
          </Wrapper>
        </RadioButton>
        <RadioButton
          id="radio-2"
          isChecked={false}
          key=".1"
          name="radio-group"
          onChange={[Function]}
          theme={
            Object {
              "input": "radio-button-input",
              "item": "radio-button",
              "label": "radio-button-label",
              "wrapper": "radio-wrapper",
            }
          }
          value="radio-2"
        >
          <Wrapper
            className="radio-button"
          >
            <Div
              className="radio-button"
            >
              <div
                className="radio-button"
              >
                <Input
                  checked={false}
                  className="radio-button-input"
                  id="radio-2"
                  onChange={[Function]}
                  type="radio"
                  value="radio-2"
                >
                  <input
                    checked={false}
                    className="radio-button-input"
                    id="radio-2"
                    onChange={[Function]}
                    type="radio"
                    value="radio-2"
                  />
                </Input>
                <label
                  className="radio-button-label"
                  htmlFor="radio-2"
                >
                  2
                </label>
              </div>
            </Div>
          </Wrapper>
        </RadioButton>
        <RadioButton
          id="radio-3"
          isChecked={false}
          key=".2"
          name="radio-group"
          onChange={[Function]}
          theme={
            Object {
              "input": "radio-button-input",
              "item": "radio-button",
              "label": "radio-button-label",
              "wrapper": "radio-wrapper",
            }
          }
          value="radio-3"
        >
          <Wrapper
            className="radio-button"
          >
            <Div
              className="radio-button"
            >
              <div
                className="radio-button"
              >
                <Input
                  checked={false}
                  className="radio-button-input"
                  id="radio-3"
                  onChange={[Function]}
                  type="radio"
                  value="radio-3"
                >
                  <input
                    checked={false}
                    className="radio-button-input"
                    id="radio-3"
                    onChange={[Function]}
                    type="radio"
                    value="radio-3"
                  />
                </Input>
                <label
                  className="radio-button-label"
                  htmlFor="radio-3"
                >
                  3
                </label>
              </div>
            </Div>
          </Wrapper>
        </RadioButton>
      </div>
    </Div>
  </Wrapper>
</RadioGroup>
`;

exports[`Radio SNAPSHOTS should render 1`] = `
<RadioGroup
  name="radio-group"
  onChange={[MockFunction]}
  value="radio-1"
>
  <Wrapper
    className="radio-wrapper"
  >
    <Div
      className="radio-wrapper"
    >
      <div
        className="radio-wrapper"
      >
        <RadioButton
          id="radio-1"
          isChecked={true}
          key=".0"
          name="radio-group"
          onChange={[Function]}
          theme={
            Object {
              "input": "radio-button-input",
              "item": "radio-button",
              "label": "radio-button-label",
              "wrapper": "radio-wrapper",
            }
          }
          value="radio-1"
        >
          <Wrapper
            className="radio-button"
          >
            <Div
              className="radio-button"
            >
              <div
                className="radio-button"
              >
                <Input
                  checked={true}
                  className="radio-button-input"
                  id="radio-1"
                  onChange={[Function]}
                  type="radio"
                  value="radio-1"
                >
                  <input
                    checked={true}
                    className="radio-button-input"
                    id="radio-1"
                    onChange={[Function]}
                    type="radio"
                    value="radio-1"
                  />
                </Input>
                <label
                  className="radio-button-label"
                  htmlFor="radio-1"
                >
                  1
                </label>
              </div>
            </Div>
          </Wrapper>
        </RadioButton>
        <RadioButton
          id="radio-2"
          isChecked={false}
          key=".1"
          name="radio-group"
          onChange={[Function]}
          theme={
            Object {
              "input": "radio-button-input",
              "item": "radio-button",
              "label": "radio-button-label",
              "wrapper": "radio-wrapper",
            }
          }
          value="radio-2"
        >
          <Wrapper
            className="radio-button"
          >
            <Div
              className="radio-button"
            >
              <div
                className="radio-button"
              >
                <Input
                  checked={false}
                  className="radio-button-input"
                  id="radio-2"
                  onChange={[Function]}
                  type="radio"
                  value="radio-2"
                >
                  <input
                    checked={false}
                    className="radio-button-input"
                    id="radio-2"
                    onChange={[Function]}
                    type="radio"
                    value="radio-2"
                  />
                </Input>
                <label
                  className="radio-button-label"
                  htmlFor="radio-2"
                >
                  2
                </label>
              </div>
            </Div>
          </Wrapper>
        </RadioButton>
        <RadioButton
          id="radio-3"
          isChecked={false}
          key=".2"
          name="radio-group"
          onChange={[Function]}
          theme={
            Object {
              "input": "radio-button-input",
              "item": "radio-button",
              "label": "radio-button-label",
              "wrapper": "radio-wrapper",
            }
          }
          value="radio-3"
        >
          <Wrapper
            className="radio-button"
          >
            <Div
              className="radio-button"
            >
              <div
                className="radio-button"
              >
                <Input
                  checked={false}
                  className="radio-button-input"
                  id="radio-3"
                  onChange={[Function]}
                  type="radio"
                  value="radio-3"
                >
                  <input
                    checked={false}
                    className="radio-button-input"
                    id="radio-3"
                    onChange={[Function]}
                    type="radio"
                    value="radio-3"
                  />
                </Input>
                <label
                  className="radio-button-label"
                  htmlFor="radio-3"
                >
                  3
                </label>
              </div>
            </Div>
          </Wrapper>
        </RadioButton>
      </div>
    </Div>
  </Wrapper>
</RadioGroup>
`;

exports[`Radio SNAPSHOTS should render controllable mode 1`] = `
<RadioGroup
  name="radio-group"
  onChange={[MockFunction]}
  value="radio-1"
>
  <Wrapper
    className="radio-wrapper"
  >
    <Div
      className="radio-wrapper"
    >
      <div
        className="radio-wrapper"
      >
        <RadioButton
          id="radio-1"
          isChecked={true}
          key=".0"
          name="radio-group"
          onChange={[Function]}
          theme={
            Object {
              "input": "radio-button-input",
              "item": "radio-button",
              "label": "radio-button-label",
              "wrapper": "radio-wrapper",
            }
          }
          value="radio-1"
        >
          <Wrapper
            className="radio-button"
          >
            <Div
              className="radio-button"
            >
              <div
                className="radio-button"
              >
                <Input
                  checked={true}
                  className="radio-button-input"
                  id="radio-1"
                  onChange={[Function]}
                  type="radio"
                  value="radio-1"
                >
                  <input
                    checked={true}
                    className="radio-button-input"
                    id="radio-1"
                    onChange={[Function]}
                    type="radio"
                    value="radio-1"
                  />
                </Input>
                <label
                  className="radio-button-label"
                  htmlFor="radio-1"
                >
                  1
                </label>
              </div>
            </Div>
          </Wrapper>
        </RadioButton>
        <RadioButton
          id="radio-2"
          isChecked={false}
          key=".1"
          name="radio-group"
          onChange={[Function]}
          theme={
            Object {
              "input": "radio-button-input",
              "item": "radio-button",
              "label": "radio-button-label",
              "wrapper": "radio-wrapper",
            }
          }
          value="radio-2"
        >
          <Wrapper
            className="radio-button"
          >
            <Div
              className="radio-button"
            >
              <div
                className="radio-button"
              >
                <Input
                  checked={false}
                  className="radio-button-input"
                  id="radio-2"
                  onChange={[Function]}
                  type="radio"
                  value="radio-2"
                >
                  <input
                    checked={false}
                    className="radio-button-input"
                    id="radio-2"
                    onChange={[Function]}
                    type="radio"
                    value="radio-2"
                  />
                </Input>
                <label
                  className="radio-button-label"
                  htmlFor="radio-2"
                >
                  2
                </label>
              </div>
            </Div>
          </Wrapper>
        </RadioButton>
        <RadioButton
          id="radio-3"
          isChecked={false}
          key=".2"
          name="radio-group"
          onChange={[Function]}
          theme={
            Object {
              "input": "radio-button-input",
              "item": "radio-button",
              "label": "radio-button-label",
              "wrapper": "radio-wrapper",
            }
          }
          value="radio-3"
        >
          <Wrapper
            className="radio-button"
          >
            <Div
              className="radio-button"
            >
              <div
                className="radio-button"
              >
                <Input
                  checked={false}
                  className="radio-button-input"
                  id="radio-3"
                  onChange={[Function]}
                  type="radio"
                  value="radio-3"
                >
                  <input
                    checked={false}
                    className="radio-button-input"
                    id="radio-3"
                    onChange={[Function]}
                    type="radio"
                    value="radio-3"
                  />
                </Input>
                <label
                  className="radio-button-label"
                  htmlFor="radio-3"
                >
                  3
                </label>
              </div>
            </Div>
          </Wrapper>
        </RadioButton>
      </div>
    </Div>
  </Wrapper>
</RadioGroup>
`;

exports[`Radio SNAPSHOTS should render controllable mode 2`] = `
<RadioGroup
  name="radio-group"
  onChange={[MockFunction]}
  value="radio-3"
>
  <Wrapper
    className="radio-wrapper"
  >
    <Div
      className="radio-wrapper"
    >
      <div
        className="radio-wrapper"
      >
        <RadioButton
          id="radio-1"
          isChecked={false}
          key=".0"
          name="radio-group"
          onChange={[Function]}
          theme={
            Object {
              "input": "radio-button-input",
              "item": "radio-button",
              "label": "radio-button-label",
              "wrapper": "radio-wrapper",
            }
          }
          value="radio-1"
        >
          <Wrapper
            className="radio-button"
          >
            <Div
              className="radio-button"
            >
              <div
                className="radio-button"
              >
                <Input
                  checked={false}
                  className="radio-button-input"
                  id="radio-1"
                  onChange={[Function]}
                  type="radio"
                  value="radio-1"
                >
                  <input
                    checked={false}
                    className="radio-button-input"
                    id="radio-1"
                    onChange={[Function]}
                    type="radio"
                    value="radio-1"
                  />
                </Input>
                <label
                  className="radio-button-label"
                  htmlFor="radio-1"
                >
                  1
                </label>
              </div>
            </Div>
          </Wrapper>
        </RadioButton>
        <RadioButton
          id="radio-2"
          isChecked={false}
          key=".1"
          name="radio-group"
          onChange={[Function]}
          theme={
            Object {
              "input": "radio-button-input",
              "item": "radio-button",
              "label": "radio-button-label",
              "wrapper": "radio-wrapper",
            }
          }
          value="radio-2"
        >
          <Wrapper
            className="radio-button"
          >
            <Div
              className="radio-button"
            >
              <div
                className="radio-button"
              >
                <Input
                  checked={false}
                  className="radio-button-input"
                  id="radio-2"
                  onChange={[Function]}
                  type="radio"
                  value="radio-2"
                >
                  <input
                    checked={false}
                    className="radio-button-input"
                    id="radio-2"
                    onChange={[Function]}
                    type="radio"
                    value="radio-2"
                  />
                </Input>
                <label
                  className="radio-button-label"
                  htmlFor="radio-2"
                >
                  2
                </label>
              </div>
            </Div>
          </Wrapper>
        </RadioButton>
        <RadioButton
          id="radio-3"
          isChecked={true}
          key=".2"
          name="radio-group"
          onChange={[Function]}
          theme={
            Object {
              "input": "radio-button-input",
              "item": "radio-button",
              "label": "radio-button-label",
              "wrapper": "radio-wrapper",
            }
          }
          value="radio-3"
        >
          <Wrapper
            className="radio-button"
          >
            <Div
              className="radio-button"
            >
              <div
                className="radio-button"
              >
                <Input
                  checked={true}
                  className="radio-button-input"
                  id="radio-3"
                  onChange={[Function]}
                  type="radio"
                  value="radio-3"
                >
                  <input
                    checked={true}
                    className="radio-button-input"
                    id="radio-3"
                    onChange={[Function]}
                    type="radio"
                    value="radio-3"
                  />
                </Input>
                <label
                  className="radio-button-label"
                  htmlFor="radio-3"
                >
                  3
                </label>
              </div>
            </Div>
          </Wrapper>
        </RadioButton>
      </div>
    </Div>
  </Wrapper>
</RadioGroup>
`;
